src/main/java/com/esri/samples/raster/raster_rendering_rule/RasterRenderingRuleSample.java

```diff
@@ -0,0 +1,203 @@
+      progressIndicator.setVisible(false);
+
+      // create drop down menu of Rendering Rules
+      renderingRuleInfoComboBox = new ComboBox<>();
+      renderingRuleInfoComboBox.setMaxWidth(260.0);
+      renderingRuleInfoComboBox.setConverter(new StringConverter<>() {
+        @Override
+        public String toString(RenderingRuleInfo renderingRuleInfo) {
+
+          return renderingRuleInfo != null ? renderingRuleInfo.getName() : "";
+        }
+
+        @Override
+        public RenderingRuleInfo fromString(String string) {
+          return null;
+        }
+      });
+
+      // create a label for the Rendering Rule info
+      Label renderingRuleInfoLabel = new Label("");
+      renderingRuleInfoLabel.setMaxWidth(220);
+      renderingRuleInfoLabel.setWrapText(true);
+
+      // add the ComboBox and Label to the controlsVBox
+      controlsVBox.getChildren().addAll(renderingRuleInfoComboBox, renderingRuleInfoLabel);
+
+      // create a Streets BaseMap
+      ArcGISMap map = new ArcGISMap(Basemap.createStreets());
+
+      // add the map to a new map view
+      mapView = new MapView();
+      mapView.setMap(map);
+
+      // create an Image Service Raster as a raster layer and add to map
+      final ImageServiceRaster imageServiceRaster = new ImageServiceRaster(IMAGE_SERVICE_RASTER_URI);
+      final RasterLayer imageRasterLayer = new RasterLayer(imageServiceRaster);
+      map.getOperationalLayers().add(imageRasterLayer);
+
+      // add event listener to loading of Image Service Raster and wait until loaded
+      imageRasterLayer.addDoneLoadingListener(() -> {
+        if (imageRasterLayer.getLoadStatus() == LoadStatus.LOADED) {
+          // zoom to extent of the raster
+          mapView.setViewpointGeometryAsync(imageServiceRaster.getServiceInfo().getFullExtent());
+
+          // get the predefined rendering rules
+          List<RenderingRuleInfo> renderingRuleInfos = imageServiceRaster.getServiceInfo().getRenderingRuleInfos();
+
+          // populate the drop down menu with the rendering rule names
+          for (RenderingRuleInfo renderingRuleInfo : renderingRuleInfos) {
+            // add the name of the rendering rule to the drop-down menu
+            renderingRuleInfoComboBox.getItems().add(renderingRuleInfo);
+          }
+          
+          // listen to selection in the drop-down menu
+          renderingRuleInfoComboBox.getSelectionModel().selectedItemProperty().addListener(o -> {
+
+            // get the requested rendering rule info from the list
+            RenderingRuleInfo selectedRenderingRuleInfo = renderingRuleInfoComboBox.getSelectionModel().getSelectedItem();
+
+            // change the label text to the rendering rule info description
+            String renderingRuleInfoDescription = selectedRenderingRuleInfo.getDescription();
+            renderingRuleInfoLabel.setText("Rule Description: \n" + renderingRuleInfoDescription);
+
+            // clear all rasters
```

Comment:
Should be `// clear all operational layers`. But that doesn't add any context. Maybe something like: `// clear raster layer using previous rendering rule`

Reply:
Opted for `// clear previous rendering rule from all operational layers`. What do you think?